packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -171,12 +232,104 @@ class ScaffoldState extends State<Scaffold> {
     final List<LayoutId>children = new List<LayoutId>();
     _addIfNonNull(children, materialBody, _Child.body);
     _addIfNonNull(children, paddedToolBar, _Child.toolBar);
-    _addIfNonNull(children, config.bottomSheet, _Child.bottomSheet);
+
+    if (_currentBottomSheet != null ||
+        (_dismissedBottomSheets != null && _dismissedBottomSheets.isNotEmpty)) {
+      List<Widget> bottomSheets = <Widget>[];
+      if (_dismissedBottomSheets != null && _dismissedBottomSheets.isNotEmpty)
+        bottomSheets.addAll(_dismissedBottomSheets);
+      if (_currentBottomSheet != null)
+        bottomSheets.add(_currentBottomSheet._widget);
+      Widget stack = new Stack(
+        bottomSheets,
+        alignment: const FractionalOffset(0.5, 1.0) // bottom-aligned, centered
+      );
+      _addIfNonNull(children, stack, _Child.bottomSheet);
+    }
+
     if (_snackBars.isNotEmpty)
       _addIfNonNull(children, _snackBars.first, _Child.snackBar);
+
     _addIfNonNull(children, config.floatingActionButton, _Child.floatingActionButton);
 
     return new CustomMultiChildLayout(children, delegate: _scaffoldLayout);
   }
 
 }
+
+class BottomSheetController {
+  const BottomSheetController._(this._widget, this.closed, this.close);
+  final Widget _widget;
+  final Future closed;
+  final VoidCallback close; // call this to close the bottom sheet
```

Comment:
OK. Unless I'm missing something I still think that just giving this class a public Future close() method would be a little nicer than a close VoidCallback and a closed Future.